src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Razor/ExtractToComponentCodeActionProvider.cs

```diff
@@ -233,4 +282,14 @@ private static bool TryGetNamespace(RazorCodeDocument codeDocument, [NotNullWhen
         // and causing compiler errors. Avoid offering this refactoring if we can't accurately get a
         // good namespace to extract to
         => codeDocument.TryComputeNamespace(fallbackToRootNamespace: true, out @namespace);
+
+    private static bool IsBlockNode(SyntaxNode node)
+        => node.Kind is
+                SyntaxKind.MarkupElement or
+                SyntaxKind.MarkupTagHelperElement or
+                SyntaxKind.CSharpCodeBlock;
+
+    private static bool IsBlockOrMarkupBlockNode(SyntaxNode node)
+        => IsBlockNode(node)
+        || node.Kind == SyntaxKind.MarkupBlock;
```

Comment:
Nit-est nit: I have no real opinion, but the indentation is inconsistent with line 132 😛